### PR TITLE
feat: Implement NFT Tab for Eth

### PR DESCRIPTION
**Pull request by Blazebrain**
- View NFTs linked to the wallet address
- Import NFTs

Issue Number (if Applicable): Fixes #

# Description

Please include a summary of the changes and which issue is fixed / feature is added. 

# Pull Request - Checklist  

- [x] Initial Manual Tests Passed
- [x] Double check modified code and verify it with the feature/task requirements
- [x] Format code
- [ ] Look for code duplication
- [ ] Clear naming for variables and methods

### CODE REVIEW

**Review by OmarHatem28 — CHANGES_REQUESTED** [review]

File: lib/entities/wallet_nft_response.dart
[the remark this answers is not in the document]
Comment (Blazebrain) [reply lib/entities/wallet_nft_response.dart:1]: Modified them to reflect this

File: lib/view_model/dashboard/nft_view_model.dart [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    log('Fetching wallet NFTs for $walletAddress');
+
+    final uri = Uri.https(
+      'deep-index.moralis.io',
+      '/api/v2.2/$walletAddress/nft',
+      {
+        "chain": "eth",
+        "format": "decimal",
+        "media_items": "false",
+        "normalizeMetadata": "true",
```
[the remark this answers is not in the document]
Comment (Blazebrain) [reply lib/view_model/dashboard/nft_view_model.dart:56]: Added documentation to the code

File: lib/router.dart [hunk 2]
```diff
@@ -607,7 +612,22 @@ Route<dynamic> createRoute(RouteSettings settings) {
...
+        builder: (_) => NFTDetailsPage(
+          nftAsset: settings.arguments as NFTAssetModel,
+          dashboardViewModel: getIt.get<DashboardViewModel>(),
+        ),
+      );
+
+    case Routes.importNFTPage:
+      return MaterialPageRoute<void>(
+        builder: (_) => ImportNFTPage(
+          nftViewModel: getIt.get<NFTViewModel>(),
```
[the remark this answers is not in the document]
Comment (Blazebrain) [reply lib/router.dart]: Adjusted it to receive the same instance of `NFTViewModel` as it's parent

File: lib/router.dart
Comment (OmarHatem28) [inline lib/router.dart:1]: please make sure you have the line width set to 100 in your IDE configuration as agreed on

File: lib/polygon/polygon.dart
Comment (OmarHatem28) [inline lib/polygon/polygon.dart:1]: unrelated to this branch and should be ignored in the polygon branch

**Review by OmarHatem28 — APPROVED** [review]

**Pull request merged**